contrib/cf-remote/cf_remote/commands.py

```diff
@@ -298,8 +299,16 @@ def spawn(platform, count, role, group_name, provider=Providers.AWS, region=None
 
         region = region or creds_data["gcp"].get("region", "europe-west1-b")
 
+    # TODO: Do we have to complicate this instead of just assuming existing VMs
+    # were created by this code and thus follow the naming pattern from this
+    # code?
+    if group_exists:
+        range_start = len(vms_info[group_key]) - 1
```

Comment:
The `- 1` seems incorrect? if there are 1 vms, the starting index should be 1, not 0, right?

Reply:
No, because we use `range(N)` if the group doesn't exist so the VM indices start with 0.

Reply:
@vpodzime First run spawns 3 VMs:
```
range(3) -> (0, 1, 2)
```

VMs have indices 0, 1 and 2.

Next run does:

```
len(vms) -> 3
```

That's the correct starting index for the next batch, we want:

```
range(len(vms), len(vms) + count) -> (3, 4, 5, ...)
```

What am I missing?

Reply:
You are right, of course. I wasn't thinking about this enough. :grin: The reason for the `-1` is this extra object that we add to the group in the JSON file:
```
    "meta": {
      "provider": "gcp",
      "region": "europe-west1-b"
    }
```
I'll make sure the code doesn't count this key in.